### PR TITLE
fetching runmode at runtime

### LINKED ISSUE #192: Fetch runmode at run time using an env variable

### PR DESCRIPTION
**Pull request by Saksham-Sharma-99**
setting runmode based on the port being used.
Fixes #192

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+
 	"github.com/astaxie/beego"
-	_ "github.com/mdg-iitr/Codephile/routers"
 	sentryhttp "github.com/getsentry/sentry-go/http"
+	_ "github.com/mdg-iitr/Codephile/conf"
+	_ "github.com/mdg-iitr/Codephile/routers"
 )
 
 func main() {
+	if os.Getenv("PORT") == "8080" {
```
Comment (dev-ritik) [inline main.go:14]: Add a new env variable `ENVIRONMENT` then using port

File: main.go [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+		fmt.Println("dev mode")
```
Comment (dev-ritik) [inline main.go:15]: Use `log` instead of `fmt`

File: main.go (hunk 1 above)
Comment (dev-ritik) [reply main.go:14]: possible values be `dev` and  `prod` for now

**Review by dev-ritik — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (Saksham-Sharma-99) [reply main.go]: done

File: main.go (hunk 2 above)
Comment (Saksham-Sharma-99) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (dev-ritik) [reply main.go]: I cant see this change

Comment (Saksham-Sharma-99) [reply main.go]: Screenshot 2021-03-27 at 3 20 08 PM

Comment (dev-ritik) [reply main.go]: > Add a new env variable `ENVIRONMENT` then using port

Replace `PORT` with  `ENVIRONMENT`

Comment (dev-ritik) [reply main.go]: > Add a new env variable `ENVIRONMENT` then using port

Replace `PORT` with  `ENVIRONMENT`

Comment (Saksham-Sharma-99) [reply main.go]: new variable in env file
Screenshot 2021-03-27 at 3 42 04 PM
___________________________
Screenshot 2021-03-27 at 3 42 32 PM

Comment (Saksham-Sharma-99) [reply main.go]: Screenshot 2021-03-27 at 3 55 14 PM

_________________________________________
new .env variable ENVIRONMENT is required to be added from now on
Screenshot 2021-03-27 at 3 42 04 PM

hopefully this is correct ?

Comment (dev-ritik) [reply main.go]: Yup

File: main.go [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 	"github.com/astaxie/beego"
-	_ "github.com/mdg-iitr/Codephile/routers"
 	sentryhttp "github.com/getsentry/sentry-go/http"
+	_ "github.com/mdg-iitr/Codephile/conf"
+	_ "github.com/mdg-iitr/Codephile/routers"
 )
 
 func main() {
+	beego.BConfig.RunMode = os.Getenv("ENVIRONMENT")
+
```
Comment (dev-ritik) [inline main.go:14]: Is this required now?

Comment (dev-ritik) [reply main.go:14]: @Saksham-Sharma-99

Comment (Saksham-Sharma-99) [reply main.go:14]: i'm removing it ..sorry got busy in some other work.

Comment (Saksham-Sharma-99) [reply main.go]: done :)

**Comment by dev-ritik** [comment]
LGTM :champagne:

**Review by dev-ritik — APPROVED** [review]

File: main.go [hunk 4]
```diff
@@ -1,10 +1,10 @@
 package main
 
 import (
-	_ "github.com/mdg-iitr/Codephile/conf"
 	"github.com/astaxie/beego"
-	_ "github.com/mdg-iitr/Codephile/routers"
 	sentryhttp "github.com/getsentry/sentry-go/http"
+	_ "github.com/mdg-iitr/Codephile/conf"
+	_ "github.com/mdg-iitr/Codephile/routers"
```
Comment (prateek2211) [inline main.go:7]: don't change the order of imports, conf package has to be loaded 1st

Comment (Saksham-Sharma-99) [reply main.go:6]: done

**Review by prateek2211 — APPROVED** [review]

**Pull request merged**